singlem/pipe.py

```diff
@@ -801,11 +817,58 @@ def _graftm_command_prefix(self, is_protein):
             cmd += "  --filter_minimum %i" % self._filter_minimum_nucleotide
 
         return cmd+' '
+    
+    def _prefilter(self, singlem_package_database, read_files):
+        '''Find all reads that match the DIAMOND database in the 
+        singlem_package database.
+        Parameters
+        ----------
+        singlem_package_database: HmmDatabase 
+            packages to search the reads for
+        read_files: list of str 
+            paths to the sequences to be searched
+        Returns
+        -------
+        path to fasta file of filtered reads
+        '''
+        dmnd = singlem_package_database.get_dmnd()
+
+        filtered_reads = []
+        prefilter_dir = os.path.join(self._working_directory, 'prefilter')
+        os.mkdir(prefilter_dir)
+        
+        for file in read_files:
+            fasta_path = os.path.join(prefilter_dir,
+                                      os.path.basename(file))
+            if fasta_path[-3:] == '.gz':
+                fasta_path = fasta_path[:-3] # remove .gz for destination files
+            fasta_path = os.path.splitext(fasta_path)[0]+'.fna'
+            
+            f = open(fasta_path, 'w+') # create tempfile in working directory
+            f.close()
+            
+            cmd = "zcat -f %s | " \
+                  "diamond blastx " \
+                  "--outfmt 6 qseqid full_qseq " \
+                  "--max-target-seqs 1 " \
+                  "--evalue 0.01 " \
+                  "--index-chunks 1 " \
+                  "--threads %i " \
+                  "--query - " \
+                  "--db %s " \
+                  "| sed -e 's/^/>/' -e 's/\\t/\\n/' > %s" % (
```

Comment:
since you are using sed here I think it needs to be added to the README as a dependency.